src/react/boosts/BoostFeed.tsx

```diff
@@ -25,16 +27,44 @@ export const BoostFeed: React.FunctionComponent = () => {
         }}
       />
       <Query query={boostFeedQuery} variables={{ search }}>
-        {({ loading, error, data }) => {
-          if (loading) {
-            return "Loading...";
-          } else if (error) {
-            return "Error: " + JSON.stringify(error);
+        {({ loading: feedQueryLoading, error: feedQueryError, data: feedQueryData }) => {
+          if (feedQueryLoading) {
+            return <LoadingMessage>Loading Boosts</LoadingMessage>;
+          } else if (feedQueryError || !feedQueryData || !feedQueryData.postsSearch) {
+            console.error("error loading Boost feed data. error:", feedQueryError, "data:", feedQueryData);
+            return "Error loading Boosts.";
           }
 
-          return data.postsSearch.posts.map((boost: any, i: number) => (
-            // `disableOwnerCheck` true because it would be slow to pull up newsroom from web3 for every single boost
-            <Boost key={i} boostId={boost.id} open={false} disableOwnerCheck={true} disableHelmet={true} />
+          return feedQueryData.postsSearch.posts.map((boostData: any, i: number) => (
+            <Query key={i} query={boostNewsroomQuery} variables={{ addr: boostData.channelID }}>
```

Comment:
Is this the only reason for the N queries on the list view? I'm going to add Channel endpoints this week, which should allow you to go from Post -> Channel -> Newsroom so you can fetch everything in a single graphql query. I'll ping you when it is complete

Reply:
@dankins that would be great thanks! On list view we only need it to query the newsroom name and charter. Then we're getting the logo from the charter. On the full boost we are using it for name, contractAddress, owner, whitelisted, charter >> image and url